### PR TITLE
Upgrade to React 15.x, Change wrapping div to wrapping span

### PR DESCRIPTION
**Pull request by anthonyjgrove**
- Upgraded to React 15.x
- Change the wrapping div to wrapping span so component can be inline.  Was having problems using this with bootstrap button groups when it was a div wrapping.
- Fix lint errors and did some ES2015 factoring
- Made adjustments to Web pack setup.

### CODE REVIEW

**Comment by anthonyjgrove** [comment]
@keppelen added fix for login bug on Chrome browser for iOS.

**Comment by keppelen** [comment]
@anthonyjgrove NICE PR :)

File: package.json [hunk 1]
```diff
@@ -1,11 +1,11 @@
 {
   "name": "react-facebook-login",
-  "version": "2.2.0",
+  "version": "3.0.0",
   "description": "A Component React for Facebook Login",
   "main": "dist/facebook-login.js",
   "scripts": {
-    "start": "webpack-dev-server --config webpack.config.js",
-    "bundle": "webpack -p --config webpack.config.dist.js",
+    "start": "NODE_ENV=development webpack-dev-server --config webpack.config.js --host $IP --port $PORT",
```
Comment (keppelen) [inline package.json]: @anthonyjgrove why you set $IP and $PORT, i think this is bad, but a need set in my bash or export IP and PORT to run project.

**Comment by anthonyjgrove** [comment]
With the set IP and PORT i get conflicts if I dont set it.  We could probably set it in the webpack setting file though with process.env.IP ||  8000 (or whatever port you want it to be).  I can play with it later.

File: src/facebook.js [hunk 2]
```diff
@@ -84,7 +84,12 @@ class FacebookLogin extends React.Component {
   };
 
   click = () => {
-    FB.login(this.checkLoginState, { scope: this.props.scope });
+    const { scope, appId } = this.props;
+    if (navigator.userAgent.match('CriOS')) {
+      window.location.href = `https://www.facebook.com/dialog/oauth?client_id=${appId}&redirect_uri=${window.location.href}&state=facebookdirect&${scope}`;
```
Comment (vovkasm) [inline src/facebook.js]: @anthonyjgrove Just wonder, if this still needed?

**Pull request merged**